util/collect_metrics_files.rb

```diff
@@ -0,0 +1,253 @@
+  service_dir = Dir.open dir_path
+  service_dir.each do |path|
+    next if path =~ /^\.$|^\.\.$/
+
+    absolute_path = "#{service_dir.path}/#{path}"
+    if File.directory?(absolute_path)
+      process_service_dir(absolute_path, service)
+      next
+    end
+    if path =~ /.json$/
+      process_json_file(absolute_path, service)
+    elsif path =~ /.tar.gz$/
+      process_tar_file(absolute_path, service)
+    end
+  end
+end
+
+# Determines if a json file is from the target window and if so, copies it to the output_dir
+#
+# @author Randell Pelak
+#
+# @param [string] file_path absolute path to json file
+# @param [string] service name of the service the file comes from
+#
+# @return [void]
+#
+# @example
+#   process_json_file(file_path, service)
+def process_json_file(file_path, service)
+  filename = File.basename(file_path)
+  time_str_with_extentsion = filename.gsub(/^#{service}-/, "")
+  time_str = time_str_with_extentsion.gsub(/.json$/, "")
+  time_obj = DateTime.parse(time_str)
+  epoch = time_obj.to_time.to_i
+  FileUtils.cp(file_path, "#{OUTPUT_DIRECTORY}/#{service}/#{filename}") if
+      epoch > START_TIME && (epoch - POLL_INTERVAL) < END_TIME
+end
+
+# Determines if a tar file contains jsons from the target window and if so,
+# either copies the whole tar file, or extracts the jsons from it to copy the relevant ones to
+# the output_dir
+#
+# @author Randell Pelak
+#
+# @param [string] file_path absolute path to tar file
+# @param [string] service name of the service the file comes from
+#
+# @return [void]
+#
+# @example
+#   process_tar_file(file_path, service)
+def process_tar_file(file_path, service)
+  filename = File.basename(file_path)
+  time_str_with_extension = filename.gsub(/^#{service}-/, "")
+  time_str = time_str_with_extension.gsub(/.tar.gz$/, "")
+  time_obj = DateTime.strptime(time_str, "%Y.%m.%d.%H.%M.%S")
+  epoch = time_obj.to_time.to_i
+  if (epoch - ARCHIVE_INTERVAL) > START_TIME && epoch < END_TIME
+    # all data in the archive is in the window
+    FileUtils.cp(file_path, "#{OUTPUT_DIRECTORY}/#{service}/#{filename}")
+  elsif epoch > START_TIME && (epoch - ARCHIVE_INTERVAL) < END_TIME
+    # part of the data in the archive may be in the window
+    Dir.mktmpdir do |dir|
+      v = (VERBOSE ? "v" : "")
+      comm = "tar xf#{v} #{file_path} -C #{dir} > /dev/null"
+      puts "Untarring archived metrics file with: #{comm}" if VERBOSE
+      system(comm)
+      process_service_dir(dir, service)
+    end
+  end
+end
+
+# Tars up the resulting metrics files for easy transport off the system
+#
+# @author Randell Pelak
+#
+# @return [void]
+#
+# @example
+#   tar_metrics_files
+def tar_metrics_files
+  v = (VERBOSE ? "v" : "")
+  comm = "tar cz#{v}f #{TAR_FILE_NAME} -C #{OUTPUT_DIRECTORY} ."
+  puts "Tarring metrics with: #{comm}" if VERBOSE
+  system(comm)
+end
+
+process_metrics_dir(METRICS_DIRECTORY)
+tar_metrics_files
+puts "Collected the following tar file with the metrics files: #{TAR_FILE_NAME}"
```

Comment:
This message is displayed even if the operation was not successful (i.e. if the specified tar file is actually a directory):
```
[root@ip-10-227-2-145 ~]# mkdir metrics
[root@ip-10-227-2-145 ~]# ruby collect_metrics_files.rb -t ~/metrics
tar (child): /root/metrics: Cannot open: Is a directory
tar (child): Error is not recoverable: exiting now
Collected the following tar file with the metrics files: /root/metrics
```